src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -173,6 +192,29 @@ private void getThePublishedCourseList(String url, CoursesForSubaccounts cfs) {
 		}
 	}
 
+	//{"errors":[{"message":"The specified resource does not exist."}],"error_report_id":479313}
+
+	private String errorHandler(HttpResponse httpResponse, String url) {
+		ObjectMapper mapper = new ObjectMapper();
+		HttpEntity entity = httpResponse.getEntity();
+		StringBuilder errMsg = new StringBuilder();
+		errMsg.append("Api call ["+url+"] has some errors they are: ");
+		try {
+			String jsonErrRes = EntityUtils.toString(entity);
+			Map<String,Object> map = new HashMap<String,Object>();
+			map = mapper.readValue(jsonErrRes,new TypeReference<HashMap<String,Object>>(){});
```

Comment:
Here is the doc I can find for the ObjectMapper's readValue call:
http://fasterxml.github.io/jackson-databind/javadoc/2.2.0/com/fasterxml/jackson/databind/ObjectMapper.html#readValue%28java.lang.String,%20com.fasterxml.jackson.core.type.TypeReference%29

public <T> T readValue(String content,
              TypeReference valueTypeRef)
            throws IOException,
                   JsonParseException,
                   JsonMappingException

Throws:
    IOException
    JsonParseException
    JsonMappingException

Note that the exceptions thrown are: IOException, JsonParseException, JsonMappingException, which are difference from the below exceptions